neoload/neoload_cli_lib/neoLoad_project.py

```diff
@@ -41,13 +42,16 @@ def zip_dir(path):
     return temp_zip
 
 
-def upload_project(path, endpoint):
+def upload_project(path, endpoint, save=None):
     filename = os.path.basename(path)
     if str(path).endswith(('.zip', '.yaml', '.yml')):
         file = open(path, "b+r")
     else:
         filename += '.zip'
         file = zip_dir(path)
+        if save is not None and save.endswith(".zip") and not os.path.exists(save):
+            copyfile(file.name, save)
```

Comment:
Per comment #discussion_r540354512, I agree and will update tomorrow.

Reply:
Used 'move' operation in zip_dir instead of copyfile. Though it still creates a TemporaryNamedFile first, it safely does the work here, then moves it to the specified save location afterwards.